clients/privacy-center/cypress/e2e/consent-events.cy.ts

```diff
@@ -0,0 +1,640 @@
+        let experience = data.items[0];
+        stubConfig({ experience });
+      });
+    });
+
+    it("should fire FidesEvents for all key interactions", () => {
+      const expectedEvents: FidesEventExpectation[] = [];
+
+      // Initialize and show banner on page load
+      expectedEvents.push(
+        { type: "FidesInitializing", detail: {} },
+        { type: "FidesInitialized", detail: {} },
+        {
+          type: "FidesUIShown",
+          detail: { extraDetails: { servingComponent: "banner" } },
+        },
+      );
+
+      // Open modal from banner button
+      cy.get("#fides-banner .fides-manage-preferences-button").click();
+      expectedEvents.push({
+        type: "FidesUIShown",
+        detail: { extraDetails: { servingComponent: "modal" } },
+      });
+
+      // Toggle first notice on then off
+      cy.get("#fides-modal .fides-toggle-input").first().click().click();
+      expectedEvents.push(
+        {
+          type: "FidesUIChanged",
+          detail: {
+            extraDetails: {
+              servingComponent: "modal",
+              trigger: {
+                type: "toggle",
+                label: "Advertising",
+                checked: true,
+              },
+              preference: {
+                key: "advertising",
+                type: "notice",
+              },
+            },
+          },
+        },
+        {
+          type: "FidesUIChanged",
+          detail: {
+            extraDetails: {
+              servingComponent: "modal",
+              trigger: {
+                type: "toggle",
+                label: "Advertising",
+                checked: false,
+              },
+              preference: {
+                key: "advertising",
+                type: "notice",
+              },
+            },
+          },
+        },
+      );
+
+      // Toggle second notice off (Analytics opt out)
+      cy.get("#fides-modal .fides-toggle-input").eq(1).click();
+      expectedEvents.push({
+        type: "FidesUIChanged",
+        detail: {
+          extraDetails: {
+            servingComponent: "modal",
+            trigger: {
+              type: "toggle",
+              label: "Analytics",
+              checked: false,
+            },
+            preference: {
+              key: "analytics_opt_out",
+              type: "notice",
+            },
+          },
+        },
+      });
+
+      // Save current preference selections
+      cy.get("#fides-modal .fides-save-button").click();
+      expectedEvents.push(
+        { type: "FidesModalClosed", detail: {} },
+        {
+          type: "FidesUpdating",
+          detail: {
+            extraDetails: {
+              consentMethod: "save",
+              // DEFER(LJ-478): Restore these checks after updating events to be more consistent
+              // servingComponent: "modal",
+              // trigger: {
+              //   type: "button",
+              //   label: "Save",
+              // },
```

Comment:
Note that our `FidesUpdating` and `FidesUpdated` events don't include `servingComponent` or any other metadata about where/how the update was triggered.

I looked into adding this context, but it was _very_ intimidating to do because of how separated the "preference update" code is from the UI hierarchy.